src/components/ProjectSearch/ProjectSearch.js

```diff
@@ -3,33 +3,63 @@
 import { DOM as dom, PropTypes, Component, createFactory } from "react";
 import { connect } from "react-redux";
 import { bindActionCreators } from "redux";
-import actions from "../actions";
-import { getSources, getProjectSearchState } from "../selectors";
-import { endTruncateStr } from "../utils/utils";
+import actions from "../../actions";
+import { getSources, getProjectSearchState } from "../../selectors";
+import { endTruncateStr } from "../../utils/utils";
 import { parse as parseURL } from "url";
-import { isPretty } from "../utils/source";
+import { isPretty } from "../../utils/source";
+import { searchSource } from "../../utils/project-search";
 import "./ProjectSearch.css";
 
-import _Autocomplete from "./shared/Autocomplete";
+import _Autocomplete from "../shared/Autocomplete";
 const Autocomplete = createFactory(_Autocomplete);
 
-function searchResults(sources) {
-  function getSourcePath(source) {
-    const { path, href } = parseURL(source.get("url"));
-    // for URLs like "about:home" the path is null so we pass the full href
-    return path || href;
+import _TextSearch from "./TextSearch";
+const TextSearch = createFactory(_TextSearch);
+
+import type { Source } from "debugger-html";
+import type { SourcesMap } from "../../reducers/sources";
+
+function getSourcePath(source: Source) {
+  if (!source.url) {
+    return "";
   }
 
-  return sources
+  const { path, href } = parseURL(source.url);
+  // for URLs like "about:home" the path is null so we pass the full href
+  return path || href;
+}
+
+function searchResults(sources: SourcesMap, query) {
+  if (false) {
```

Comment:
this looks weird, because we will never enter this loop. should it be deleted, or updated to be meaningful?